lms/templates/dashboard/_dashboard_entitlement_actions.html

```diff
@@ -0,0 +1,47 @@
+<%page args="course_overview, entitlement, dashboard_index" expression_filter="h"/>
+
+<%!
+from django.utils.translation import ugettext as _
+from django.core.urlresolvers import reverse
+%>
+
+<%
+dropdown_id = "entitlement-actions-dropdown-{}".format(dashboard_index)
+dropdown_btn_id = "entitlement-actions-dropdown-btn-{}".format(dashboard_index)
+%>
+
+<div class="entitlement-actions-wrapper">
+  ## id, data-dropdown-selector, and data-dropdown-button-selector must be defined for compatibility with
+  ## lms/static/js/dashboard/dropdown.js
+  <button id="${dropdown_btn_id}"
+          class="entitlement-action entitlement-action-more js-entitlement-action-more"
+          type="button"
+          aria-haspopup="true"
+          aria-expanded="false"
+          aria-controls="${dropdown_id}"
+          data-dropdown-selector="#${dropdown_id}"
+          data-dropdown-button-selector="#${dropdown_btn_id}">
+    <span class="sr">${_('Course options for {courseName}').format(courseName=course_overview.display_name_with_default)}</span>
+    <span class="fa fa-cog" aria-hidden="true"></span>
+  </button>
+
+  <div id="${dropdown_id}" class="entitlement-actions-dropdown" tabindex="-1">
+    <ul class="entitlement-actions-dropdown-list" aria-label="${_('Available Actions')}" role="menu">
+      <li class="entitlement-actions-item" role="menuitem">
+        ## href, id, rel, and data-model-close-button-selector must be defined for compatibility with lms/static/js/leanModal.js
+        ## data-dropdown-selector and data-dropdown-button-selector must be defined for compatibility with lms/static/js/dashboard/dropdown.js
+        ## class="js-entitlement-action-unenroll" must be set for compatibility with lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js
+        <a href="#entitlement-unenrollment-modal" id="entitlement-action-unenroll-${dashboard_index}" class="entitlement-action js-entitlement-action-unenroll" rel="leanModal"
```

Comment:
If this ```a``` just opens a modal then it should really be a ```button``` from an a11y/HTML perspective.

Reply:
I believe https://github.com/edx/edx-platform/blob/master/lms/static/js/leanModal.js#L9 requires this to be an `<a>`, unfortunately, as it relies on the `href` attribute to find the modal in the DOM. Curiously, at the end of that file it binds events to `<button>` objects though, so I'm not sure how that works without `href`.

Reply:
Thanks for the clarification.